apis/v1alpha1/httproute_types.go

```diff
@@ -432,6 +432,24 @@ const (
 // HTTPRequestHeaderFilter defines configuration for the RequestHeaderModifier
 // filter.
 type HTTPRequestHeaderFilter struct {
+	// Set overwrites the request with the given header (name, value)
+	// before the action.
+	//
+	// Input:
+	//   GET /foo HTTP/1.1
+	//   My-Header: ABC
+	//
+	// Config:
+	//   set: {"my-header": "XYZ"}
+	//
+	// Output:
+	//   GET /foo HTTP/1.1
+	//   my-header: XYZ
+	//
+	// Support: Extended
+	// +optional
+	Set map[string]string `json:"set,omitempty"`
```

Comment:
+1 to this change.
One concern. Can most implementations differentiate between addition and override?

Reply:
From what I can tell, adding headers is pretty widely supported among other implementations, but setting/overwriting headers is certainly trickier. HAProxy has a very clear differentiation that would match this config closely (add-header and set-header). The distinction is less clear with nginx (add_header and proxy_set_header but defined in different modules), and I think you might need a custom filter to overwrite a header with envoy. 

With all that said, I think this is a good addition for the implementations that can support it, and the `extended` support level feels appropriate.